### DISCUSSION TITLE
postgREST installation, connectivity with postgres database issue and creation of API  -Reg

### DISCUSSION THREAD

**Discussion by sureshbellamkonda96** [PostgREST/postgrest#3649] [Q&A] [answered]
Hi,

I had postgres server with database in my local system, i want to create API for my database tables for that i was downloaded postgREST binaries while am trying to install postgREST in my system then facing some challenges.

Did: 1.Download and Extracted the postgREST files at directory.
         2. Created the Config file at same location with details of database, port.
         
Challenges/Error:

While i was trying in my system command prompt with location of postgREST, config file i am getting this error:
postgrest: Fatal Error {fatalErrorMessage = "Error in config postgrest.conf: open File: does not exist (No such file or directory)"}

### Environment

* PostgreSQL version: 15
* PostgREST version: postgrest-v12.2.1-windows-x64
* Operating system: windows

### Description of issue

Could please provide me directions for how to install postgREST, Creation of config file and how to create API by using my database.
   (Creation of API is mandatory for us please help on this)

 Hope i can get response soon.
  Thanks in advance.

Regards,
Suresh.

**Comment by wolfgangwalther** [comment]
How did you start PostgREST? The exact command?

How does your `postgrest.conf` file look like? Please **remove all secrets** from it when posting here (like passwords and jwt-secret etc.)

**Reply by sureshbellamkonda96** [reply]
in notepad i have wrote this below, and save at postgrest directory  
db-uri = "postgres://username:password@localhost:****/Database"
db-schema = "public"

**Reply by wolfgangwalther** [reply]
> How did you start PostgREST? The exact command?

**Reply by sureshbellamkonda96** [reply]
postgrest postgrest.conf

**Reply by sureshbellamkonda96** [reply]
C:\Users\ssssss\Downloads\artifacts\postgrest-windows-x64>postgrest postgrest.conf

**Reply by wolfgangwalther** [reply]
Are you sure you created `postgrest.conf` and not accidentally `postgrest.conf.txt` or so?

**Reply by wolfgangwalther** [reply]
I.e. run `dir` in the same folder and see what it prints.

**Reply by sureshbellamkonda96** [reply]
> Are you sure you created `postgrest.conf` and not accidentally `postgrest.conf.txt` or so?
simply i have taken notepad and filled with database, port, etc.. then save with name as postgrest.conf that's it.
 but it was not created automatically....

**Reply by wolfgangwalther** [reply, marked as answer]
> simply i have taken notepad and filled with database, port, etc.. then save with name as postgrest.conf that's it.

That's a very strong indication that notepad has added `.txt` to it at the end. This will not be displayed in the file manager by default, but it's there.

You might want to call `postgrest postgrest.conf.txt` to try out.

**Comment by sureshbellamkonda96** [comment]
when i ran postgrest postgrest.conf.txt i got like below one, 
but unable to run postgrest can you pl guide me....
 >postgrest postgrest.conf.txt
 Starting PostgREST 12.2.1 (******)...
 Listening on port ****
Attempting to connect to the database...
Successfully connected to PostgreSQL 15.4, 
 Config reloaded

**Reply by wolfgangwalther** [reply]
This means it works fine.

**Comment by wolfgangwalther** [comment]
The problem here is about loading the config file. This problem has been solved. Closing as resolved.

**Reply by sureshbellamkonda96** [reply]
Thank you very much @wolfgangwalther
